src/middlewared/middlewared/alert/source/m_series_nvdimm_firmware_freebsd.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding=utf-8 -*-
+import logging
+
+from middlewared.alert.base import Alert, AlertClass, AlertCategory, AlertLevel, AlertSource
+from middlewared.service_exception import CallError
+
+logger = logging.getLogger(__name__)
+
+
+class TrueNASMNVDIMMFirmwareVersionAlertClass(AlertClass):
+    category = AlertCategory.HARDWARE
+    level = AlertLevel.EMERGENCY
+    title = "Invalid NVDIMM Firmware Version"
+    text = (
+        "You are using an invalid NVDIMM firmware version %(version)s which can produce data loss. "
+        "Please contact support immediately."
+    )
+
+    products = ("ENTERPRISE",)
+    hardware = True
+
+
+class TrueNASMNVDIMMFirmwareVersionAlertSource(AlertSource):
+    products = ("ENTERPRISE",)
+
+    async def check(self):
+        if (await self.middleware.call("truenas.get_chassis_hardware")).startswith("TRUENAS-M"):
```

Comment:
I'd rather use `failover.hardware == 'ECHOWARP'` instead considering we now sell truenas minis which can be flashed `truenas-mini` which do not have nvdimms. Also, failover.hardware runs `getencstat` on the ses devices and checks the name returned in the firmware, which should be more predictable.

```suggestion
        if await self.middleware.call("failover.hardware") == "ECHOWARP":
```

Reply:
I don't like the idea of hardcoding magic words like `ECHOWARP`. Will truenas-mini identify like TRUENAS-MINI? Then we probably have other failing bits of code that use `startswith("TRUENAS-M")` like:
```
    if product.startswith('TRUENAS-M') and current_vers.major == minimum_vers.major:
        return [
            'hint.ntb_hw.0.split=1',
            'hint.ntb_hw.0.config="ntb_pmem:1:4:0,ntb_pmem:1:4:0,ntb_transport"'
        ]
```

Reply:
Also I found this
```
is_m_series = baseboard_manufacturer == "Supermicro" and failover_hardware == "ECHOWARP"
```

Reply:
`ECHOWARP` is not a magic string. It's the product name we use to describe the m-series head-unit chassis. As a matter of fact, if that detection fails, then HA will not work. The Z and X series platforms have their own names as well.

Yes, the new truenas minis will be tagged with `TRUENAS-MINI` so the other code bits will have an opportunity to fail. I'll open a ticket about it.